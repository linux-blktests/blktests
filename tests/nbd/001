#!/bin/bash
#
# Test nbd device resizing.
#
# Copyright (C) 2018 Josef Bacik
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

. common/nbd

DESCRIPTION="resize a connected nbd device"
QUICK=1

requires() {
	_have_nbd && _have_program parted && _have_src_program nbdsetsize
}

test() {
	echo "Running ${TEST_NAME}"
	_start_nbd_server
	nbd-client -N export localhost /dev/nbd0 > /dev/null 2>&1
	parted -s /dev/nbd0 print | grep Disk
	lsblk -n /dev/nbd0

	echo "Setting size to 1gib"
	src/nbdsetsize /dev/nbd0 1073741824

	lsblk -n /dev/nbd0
	parted -s /dev/nbd0 print | grep Disk

	nbd-client -d /dev/nbd0 > /dev/null 2>&1
	_stop_nbd_server
}
