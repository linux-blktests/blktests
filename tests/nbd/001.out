Running nbd/001
Disk /dev/nbd0: 10.7GB
nbd0  43:0    0  10G  0 disk 
Setting size to 1gib
nbd0  43:0    0   1G  0 disk 
Disk /dev/nbd0: 1074MB
